extension/packagemanagerextension/serverextension/envmanager.py

```diff
@@ -0,0 +1,329 @@
+        kdir = '.local/share/jupyter/kernels/' + env
+        os.remove(kdir + '/kernel.json')
+        os.rmdir(kdir)
+        return self.clean_conda_json(output)
+
+    def project_info(self, directory):
+        env = ""
+        names = []
+        packagesMD = {}
+        directory = str(directory) + ".swanproject"
+        try:
+            val = yaml.load(open(directory))
+            env = val['ENV']
+            packagesMD = val['PACKAGE_INFO']
+        except:
+            return {'error': "Can't find project"}
+
+        for item in packagesMD:
+            names.append(item.get('name'))
+
+        output = self._execute('conda list --no-pip --json -n', env)
+        data = self.clean_conda_json(output)
+        if 'error' in data:
+            # we didn't get back a list of packages, we got a dictionary with
+            # error info
+            return data
+
+        return {
+            "env": env,
+            "packages": [pkg_info_status(package, names) for package in data]
+        }
+
+    def export_env(self, env):
+        return self._execute('conda list -e -n', env)
+
+    def clone_env(self, env, name):
+        output = self._execute('conda create -y -q --json -n', name,
+                               '--clone', env)
+        return self.clean_conda_json(output)
+
+    def create_project(self, directory, type):
+        name = uuid.uuid1()
+        name = 'swanproject-' + str(name)
+        data = {'ENV': name}
+
+        folder = directory[:-1].split('/')[-1]
+
+        if not os.path.exists(directory):
+            os.makedirs(directory)
+
+        # if os.path.isfile(directory + '.swanproject'):
+        #     res = {'error': 'This directory alreday contains a project'}
+        #     return res
+
+        packages = package_map[type]
+        output = self._execute('conda create -y -q --json -n', name,
+                               *packages.split(" "))
+        packages = self._execute('conda list --no-pip --json -n', name)
+        packages = self.clean_conda_json(packages)
+        op = self.clean_conda_json(output)
+        tp = json.dumps(op)
+        js = json.loads(tp)
+        kerneljson = {
+            "argv": [js['prefix'] + "/bin/python",   "-m",
+                     "ipykernel_launcher",
+                     "-f",
+                     "{connection_file}"],
+            "display_name": "Python (" + folder + ")",
+            "language": "python"
+        }
+        kdir = '.local/share/jupyter/kernels/' + name
+        if not os.path.exists(kdir):
+            os.makedirs(kdir)
+        with open(kdir + '/kernel.json', 'w') as fp:
+            json.dump(kerneljson, fp)
+
+        data['PACKAGE_INFO'] = packages
+        with open(directory + '.swanproject', 'w') as outfile:
+            yaml.dump(data, outfile, default_flow_style=False)
+        return op
+
+    def env_packages(self, env):
+        output = self._execute('conda list --no-pip --json -n', env)
```

Comment:
what does --no-pip do? remove packages installed by pip? shouldn't we also preserve those?

Reply:
That will include the list of all the global packages made available through pip. Is that okay?

Reply:
Global, like in what? If the packages that come from pip are only in the environment, then we should include them. Or global means, global pip installation, available in all the environments?